ts/indicators/cci.src.ts

```diff
@@ -0,0 +1,186 @@
+            public data: Array<CciIndicatorPoint>;
+            public getValues(
+                series: Series,
+                params: CciIndicatorParamsOptions
+            ): (boolean|IndicatorValuesObject);
+            public options: CciIndicatorOptions;
+            public yData: Array<Array<number>>;
+        }
+
+        interface CciIndicatorOptions extends SmaIndicatorOptions {
+            params?: CciIndicatorParamsOptions;
+        }
+
+        interface CciIndicatorParamsOptions extends SmaIndicatorParamsOptions {
+            // declared for inheritance
+        }
+
+        class CciIndicatorPoint extends SmaIndicatorPoint {
+            public series: CciIndicator;
+        }
+
+        interface SeriesTypesDictionary {
+            cci: typeof CciIndicator;
+        }
+    }
+}
+
+import U from '../parts/Utilities.js';
+var isArray = U.isArray;
+
+var seriesType = H.seriesType;
+
+/* eslint-disable valid-jsdoc */
+// Utils:
+/**
+ * @private
+ */
+function sumArray(array: Array<number>): number {
+    return array.reduce(function (prev, cur): number {
+        return prev + cur;
+    }, 0);
+}
+
+/**
+ * @private
+ */
+function meanDeviation(arr: Array<number>, sma: number): number {
+    var len = arr.length,
+        sum = 0,
+        i: number;
+
+    for (i = 0; i < len; i++) {
+        sum += Math.abs(sma - (arr[i]));
+    }
+
+    return sum;
+}
+/* eslint-enable valid-jsdoc */
+
+/**
+ * The CCI series type.
+ *
+ * @private
+ * @class
+ * @name Highcharts.seriesTypes.cci
+ *
+ * @augments Highcharts.Series
+ */
+seriesType<Highcharts.CciIndicator>(
+    'cci',
+    'sma',
+    /**
+     * Commodity Channel Index (CCI). This series requires `linkedTo` option to
+     * be set.
+     *
+     * @sample stock/indicators/cci
+     *         CCI indicator
+     *
+     * @extends      plotOptions.sma
+     * @since        6.0.0
+     * @product      highstock
+     * @optionparent plotOptions.cci
+     */
+    {
+        params: {
+            period: 14
+        }
+    },
+    /**
+     * @lends Highcharts.Series#
+     */
+    {
+        getValues: function (
+            series: Highcharts.CciIndicator,
+            params: Highcharts.CciIndicatorParamsOptions
+        ): (boolean|Highcharts.IndicatorValuesObject) {
+            var period = (params.period as any),
+                xVal = (series.xData as any),
+                yVal: Array<Array<number>> = series.yData,
+                yValLen: number = yVal ? yVal.length : 0,
+                TP: Array<number> = [],
+                periodTP: Array<number> = [],
+                range = 1,
+                CCI: Array<Array<number>> = [],
+                xData: Array<number> = [],
+                yData: Array<number> = [],
+                CCIPoint: number,
+                p: Array<number>,
+                len: number,
+                smaTP: number,
+                TPtemp: number,
+                meanDev: number,
+                i: (number|undefined);
```

Comment:
Try without `undefined` and remove the `any` casts in the for-loop. For this the `period` variable has to be explicitly set to type `number`.